### ISSUE TITLE
Headers are shown multiple times

### ISSUE THREAD

**Issue by thelong1EU** [davideas/FlexibleAdapter#355]
![ezgif-3-eb77d37d19](https://cloud.githubusercontent.com/assets/9283414/25775910/1717ecd8-32b9-11e7-901f-a2a43691bc95.gif)
I switched to this wonderful lib. I spend whole day reading docs. But still I have this issue. Headers are displays above every item.

As always the code is here:
https://github.com/thelong1EU/FlipViewDemo

**Comment by davideas** [comment]
@thelong1EU, `UserHeader` must be the _same_ instance for each section, so you should NOT create new header object for each sectionable in your use case.

**Comment by thelong1EU** [comment]
I see. I thought that the equals method deals with it.

**Comment by thelong1EU** [comment]
I changed it in the mock app and it works. But on my real app it doesn't. I get the same result.
Where are you from?

```
public ArrayList<BookItem> list() {
        ArrayList<BookItem> bookItem = new ArrayList<>();
        String[] columns = {COLUMN_LOCATION, COLUMN_USER, COLUMN_BOOK};
        UserHeader header = null;
        Cursor cursor = database.query(TABLE_TITLE, columns, null, null, null, null, COLUMN_USER + " ASC");
        if (cursor != null && cursor.moveToFirst()) {
            do {
                String databaseLocation = cursor.getString(cursor.getColumnIndex(COLUMN_LOCATION));
                User user = User.getFromJson(cursor.getString(cursor.getColumnIndex(COLUMN_USER)));
                Book book = Book.getFromJson(cursor.getString(cursor.getColumnIndex(COLUMN_BOOK)));

                talk.setDatabaseLocation(databaseLocation);
                user.setDatabaseLocation(databaseLocation);

                if (header == null) header = new UserHeader(user);
                UserHeader newHeader = new UserHeader(user);
                if (!header.equals(newHeader)) header = newHeader;

                BookItem item = new BookItem(book, header);
                bookItem.add(item);
            } while (cursor.moveToNext());
            cursor.close();
        }

        log.d(talkItems.get(0).getHeader() == talkItems.get(5).getHeader());//returns true
        log.d(talkItems.get(1).getHeader() == talkItems.get(4).getHeader());//returns true
        log.d(talkItems.get(2).getHeader() == talkItems.get(3).getHeader());//returns true
        return bookItems;
    }
```

**Comment by thelong1EU** [comment]
I think I found the reason. In the FlexibleAdapter there is the method showAllHeadersWithReset(). Here you check if the headers are the same instance. Why not try equals?

**Comment by davideas** [comment]
Hi @thelong1EU, I didn't think about Parcelable, sometimes we write a feature without seeing all the user points of views for development, some can be missed. The improvement can be done.

I'm Italian living in Brussels. for the beer I will check about donations, because you are not the first, neither second that would like to, I appreciate it :-D

**Comment by thelong1EU** [comment]
Added fix to #357, so I'm closing this.
Thanks again.

**Comment by thelong1EU** [comment]
Added fix to #357, so I'm closing this.
Thanks again.
